hpOneView/resources/resource.py

```diff
@@ -186,7 +186,7 @@ def delete(self, resource, force=False, timeout=-1, custom_headers=None):
 
         task, body = self._connection.delete(uri, custom_headers=custom_headers)
 
-        if not task:
+        if not task or 'taskState' not in task:
```

Comment:
Please, disconsider the last comment.
Reviewing better, the problem is in connection, connection should not return a task that is not a task.

Now I suggest you to change the function: connection.__do_rest_call 
If location does not exist, we need to check if body is a task before setting it, otherwise, we need to return None.